editor/views/files.py

```diff
@@ -115,10 +120,26 @@ def handle_on_click():
                 file = file_from_url(file_type, url, names)
             elif uploaded_file:
                 file = file_from_upload(file_type, uploaded_file, names)
-            # todo: handle manually created resource.
             else:
-                st.toast("Please, import either a local file or a URL.", icon="❌")
-                return
+                resource_type = st.session_state[_MANUAL_RESOURCE_TYPE_KEY]
+                needs_sha256 = resource_type == FILE_OBJECT
+
+                name = st.session_state[_MANUAL_NAME_KEY]
+                description = st.session_state[_MANUAL_DESCRIPTION_KEY]
+                sha256 = st.session_state[_MANUAL_SHA256_KEY] if needs_sha256 else None
+
+                if not name or (needs_sha256 and not sha256):
+                    # Some required fields are empty.
+                    st.toast(
+                        "Please, import either a local file, provide a download URL, or"
```

Comment:
Is the error message relevant to the error? `Please, fill in all required fields: ...`?

Reply:
Modified the error message. Basically we show the error if neither uploaded file, nor URL, nor the form with the manual input is filled in so that we can actually add a Resource. Probably would be better to have a dedicated error message for each "option".

Also, do we want to maybe not clear the form if it wasn't handled successfully? This way if the user e.g. inputs name, description and clicks "Upload" we don't delete everything but they can just add the SHA256.